src/main/scala/mg7/pipeline.scala

```diff
@@ -0,0 +1,209 @@
+
+      listChunks( blastChunksS3Prefix(sampleId) )
+        .zipWithIndex
+        .map { case (chunkS3Obj, n) =>
+
+          DataMapping(sampleId, assignDataProcessing(parameters))(
+            remoteInput = Map(
+              data.blastChunk -> S3Resource(chunkS3Obj)
+            ),
+            remoteOutput = Map(
+              data.lcaChunk -> S3Resource(lcaAssignS3Prefix(sampleId) / s"${sampleId}.lca.${n}.csv"),
+              data.bbhChunk -> S3Resource(bbhAssignS3Prefix(sampleId) / s"${sampleId}.bbh.${n}.csv")
+            )
+          )
+        }
+    }
+
+    lazy val merge = inputSamples.keys.toList.map { case sampleId =>
+
+      DataMapping(sampleId, mergeDataProcessing)(
+        remoteInput = Map(
+          data.blastChunksFolder -> S3Resource(blastChunksS3Prefix(sampleId)),
+          data.blastNoHitsFolder -> S3Resource(blastNoHitsS3Prefix(sampleId)),
+          data.lcaChunksFolder   -> S3Resource(lcaAssignS3Prefix(sampleId)),
+          data.bbhChunksFolder   -> S3Resource(bbhAssignS3Prefix(sampleId))
+        ),
+        remoteOutput = Map(
+          data.blastResult -> S3Resource(outputS3Folder(sampleId, "merge") / s"${sampleId}.blast.csv"),
+          data.blastNoHits -> S3Resource(outputS3Folder(sampleId, "merge") / s"${sampleId}.no-hits.fa"),
+          data.lcaCSV      -> S3Resource(outputS3Folder(sampleId, "merge") / s"${sampleId}.lca.csv"),
+          data.bbhCSV      -> S3Resource(outputS3Folder(sampleId, "merge") / s"${sampleId}.bbh.csv")
+        )
+      )
+    }
+
+    lazy val count: List[DataMapping[countDataProcessing.type]] = merge.map { case mergeDM =>
+      val sampleId = mergeDM.label
+
+      def outputFor(d: FileData): (FileData, S3Resource) = {
+        d -> S3Resource(outputS3Folder(sampleId, "count") / s"${sampleId}.${d.baseName}.csv")
+      }
+
+      DataMapping(sampleId, countDataProcessing)(
+        remoteInput = Map(
+          lookup(data.lcaCSV, mergeDM.remoteOutput),
+          lookup(data.bbhCSV, mergeDM.remoteOutput)
+        ),
+        remoteOutput = Map(
+          outputFor(data.lca.direct.absolute),
+          outputFor(data.lca.accum.absolute),
+          outputFor(data.lca.direct.relative),
+          outputFor(data.lca.accum.relative),
+          outputFor(data.bbh.direct.absolute),
+          outputFor(data.bbh.accum.absolute),
+          outputFor(data.bbh.direct.relative),
+          outputFor(data.bbh.accum.relative)
+        )
+      )
+    }
+  }
+
+  case object split  extends Loquat(splitConfig,  splitDataProcessing(parameters))(dataMappings.split)
+  case object blast  extends Loquat(blastConfig,  blastDataProcessing(parameters))(dataMappings.blast)
+  case object assign extends Loquat(assignConfig, assignDataProcessing(parameters))(dataMappings.assign)
+  case object merge  extends Loquat(mergeConfig,  mergeDataProcessing)(dataMappings.merge)
+  case object count  extends Loquat(countConfig,  countDataProcessing)(dataMappings.count)
+}
```

Comment:
Well this is the whole point of the pipeline: you provide the parameters and get all the loquats defined. Loquat has to be an object and I don't see a way to define them outside.